Add RKE2+EC2 docs (#296)
rancher/turtles-docs: docs/next/modules/en/pages/user/clusters.adoc

Reviewer: Nit: Unrelated to the changes of this PR but can we order the tabs under the `Create Your Cluster Definition` section to be sorted alphabetically?

Author: let's do it after all clusterclasses are added

[tabs]
======

> AWS EKS::

Reviewer: Should this just say `AWS`? Because below we cover both EKS and "bring your own AWS infra" provider options.

======
AWS EC2 RKE2::
+

> * Following installation guide was moved here xref:../user/clusterclass.adoc[here]

Reviewer 2: no need to mention a doc move, let's instead use:
```suggestion
* You can follow the installation guide xref:../user/clusterclass.adoc[here]
```

AWS EC2 Kubeadm::
+

> * Following installation guide was moved here xref:../user/clusterclass.adoc[here]

Reviewer 2: ```suggestion
* You can follow the installation guide xref:../user/clusterclass.adoc[here]
```
